tests/catch/unit/memory/hipFree.cc

```diff
@@ -0,0 +1,580 @@
+    auto flag = GENERATE(hipHostRegisterDefault, hipHostRegisterPortable, hipHostRegisterMapped);
+    HIP_CHECK(hipHostRegister((void*)hostPtr, sizeof(char), flag));
+    HIP_CHECK_ERROR(freeStuff(hostPtr, HostFree), hipErrorInvalidValue);
+  }
+}
+
+#if HT_NVIDIA
+TEST_CASE("Unit_hipFreeNegativeArray") {
+  DriverContext ctx;
+  hipArray_t arrayPtr{};
+  hiparray cuArrayPtr{};
+
+  SECTION("InvalidPtr") {
+    arrayPtr = static_cast<hipArray*>(malloc(sizeof(char)));
+
+    SECTION("ArrayFree") {
+      HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayFree), hipErrorContextIsDestroyed);
+    }
+    SECTION("ArrayDestroy") {
+      HIP_CHECK_ERROR(freeStuff(cuArrayPtr, ArrayDestroy), hipErrorInvalidResourceHandle);
+    }
+    free(arrayPtr);
+  }
+  SECTION("NullPtr") {
+    arrayPtr = nullptr;
+    SECTION("ArrayFree") { HIP_CHECK(freeStuff(arrayPtr, ArrayFree)); }
+    SECTION("ArrayDestroy") {
+      HIP_CHECK_ERROR(freeStuff(cuArrayPtr, ArrayDestroy), hipErrorInvalidResourceHandle);
+    }
+  }
+}
+#else
+
+// Freeing a invalid pointer with array
+TEST_CASE("Unit_hipFreeNegativeArray") {
+  hipArray_t arrayPtr{};
+
+  SECTION("InvalidPtr") {
+    arrayPtr = static_cast<hipArray*>(malloc(sizeof(char)));
+
+    SECTION("ArrayFree") {
+      HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayFree), hipErrorContextIsDestroyed);
+    }
+    SECTION("ArrayDestroy") {
+      HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayDestroy), hipErrorContextIsDestroyed);
+    }
+  }
+
+  SECTION("NullPtr") {
+    arrayPtr = nullptr;
+    SECTION("ArrayFree") { HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayFree), hipErrorInvalidValue); }
+    SECTION("ArrayDestroy") {
+      HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayDestroy), hipErrorInvalidValue);
+    }
+  }
+}
+
+#endif
+
+
+TEST_CASE("Unit_hipFreeDoubleDevice") {
+  size_t width = GENERATE(32, 512, 1024);
+  char* ptr{};
+  size_t size_mult = width;
+  HIP_CHECK(hipMalloc(&ptr, sizeof(char) * size_mult));
+
+  HIP_CHECK(freeStuff(ptr, DevFree));
+  HIP_CHECK_ERROR(freeStuff(ptr, DevFree), hipErrorInvalidValue);
+}
+TEST_CASE("Unit_hipFreeDoubleHost") {
+  size_t width = GENERATE(32, 512, 1024);
+  char* ptr{};
+  size_t size_mult = width;
+
+  HIP_CHECK(hipHostMalloc(&ptr, sizeof(char) * size_mult));
+
+  HIP_CHECK(freeStuff(ptr, HostFree));
+  HIP_CHECK_ERROR(freeStuff(ptr, HostFree), hipErrorInvalidValue);
+}
+TEST_CASE("Unit_hipFreeDoubleArrayFree") {
+#if HT_NVIDIA
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-120");
+  return;
+#endif
+  size_t width = GENERATE(32, 512, 1024);
+  size_t height = GENERATE(0, 32, 512, 1024);
+  hipArray_t arrayPtr{};
+  hipExtent extent{};
+  extent.width = width;
+  extent.height = height;
+  hipChannelFormatDesc desc = hipCreateChannelDesc<char>();
+
+  HIP_CHECK(hipMallocArray(&arrayPtr, &desc, extent.width, extent.height, hipArrayDefault));
+
+  HIP_CHECK(freeStuff(arrayPtr, ArrayFree));
+  HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayFree), hipErrorContextIsDestroyed);
+}
+#if HT_NVIDIA
```

Comment:
why is this test both removed with the preprocessor and skipped?

Reply:
Some inconsistent failures are being looked into this test case, which might or might not fail, keeping test as skipped for now.